### PR TITLE
vendor: Remove gojay

### PR DESCRIPTION
**Pull request by michi-covalent**
This PR removes the WithJSONEncoder printer option and gojay dependency
since the observe command is always configured to use the standard JSON
encoder.

Signed-off-by: Michi Mutsuzaki <michi@isovalent.com>

### CODE REVIEW

**Comment by glibsm** [comment]
The fact that no tests broke means one of two things:
1. There weren't tests 😄 
2. They still pass

I'm assuming the latter?

**Comment by michi-covalent** [comment]
> The fact that no tests broke means one of two things:
> 
>     1. There weren't tests 😄
> 
>     2. They still pass
> 
> 
> I'm assuming the latter?

definitely the latter. the observe command never used gojay encoder :)

File: vendor/modules.txt [hunk 1]
```diff
@@ -63,7 +62,6 @@ github.com/davecgh/go-spew/spew
 github.com/fatih/color
 # github.com/francoispqt/gojay v1.2.13
 ## explicit
```
Comment (Rolinh) [inline vendor/modules.txt]: Mmh, this shouldn't be an explicit dependency anymore. Please, re-run `go mod tidy && go mod vendor`.

Comment (michi-covalent) [reply vendor/modules.txt]: done. somehow i always keep forgetting this :(

**Comment by michi-covalent** [comment]
will remove the interface

**Pull request merged**